src/unpkg-plugin/UnpkgPluginImpl.ts

```diff
@@ -0,0 +1,74 @@
+// Built by modifying https://github.com/gdelmas/typedoc-plugin-sourcefile-url
+
+import * as path from 'path';
+import * as fs from 'fs';
+import { Component } from 'typedoc/dist/lib/utils/component';
+import { ConverterComponent } from 'typedoc/dist/lib/converter/components';
+import { Converter } from 'typedoc/dist/lib/converter/converter';
+import { Context } from 'typedoc/dist/lib/converter/context';
+import { SourceReference } from 'typedoc/dist/lib/models/sources/file';
+import { Options } from 'typedoc/dist/lib/utils/options/options';
+import { sync as pkgUp } from 'pkg-up';
+import { URL } from 'url';
+import { dirname, relative } from 'path';
+
+@Component({ name: 'unpkg-plugin' })
+export class UnpkgPlugin extends ConverterComponent {
+  private pkgUpCache = new Map<string, string | null>();
+  private pkgUp(cwd: string) {
+    let cacheEntry = this.pkgUpCache.get(cwd);
+    if (cacheEntry !== undefined) return cacheEntry;
+    cacheEntry = pkgUp({ cwd });
+    this.pkgUpCache.set(cwd, cacheEntry);
+    return cacheEntry;
+  }
+
+  public initialize(): void {
+    this.listenTo(this.owner, Converter.EVENT_BEGIN, this.onBegin);
+  }
+
+  private onBegin(): void {
+    // register handler
+    this.listenTo(this.owner, Converter.EVENT_RESOLVE_END, this.onEndResolve);
```

Comment:
Why is the event registration split up?

Reply:
Same reason as above, it was copy-pasted from https://github.com/gdelmas/typedoc-plugin-sourcefile-url